.github/workflows/build_and_test.yml

```diff
@@ -312,18 +312,17 @@ jobs:
       if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch'
       timeout-minutes: 20
       run: |
-        pip install poetry==1.6.1
+        pip install poetry==1.8.1
```

Comment:
Is it important that it be exactly this version of `poetry` or can we just assume the latest version will work?

Reply:
I updated it so that I could use `package-mode=false` in `pyproject.toml` this clarifies that we're only using poetry to install the docs dependencies. We are not using it for package management here.

Seems this isn't available in 1.6.1.

Reply:
So we can probably make it `>=1.8.1`? Less likely to have dependency conflicts in the future if we do that.

Reply:
Good idea, updated [dd03ddd](https://github.com/CQCL/tket/pull/1609/commits/dd03dddb24a9a0ee65982c1c3fa5656c97ff085c)